addons/l10n_es_edi_facturae/models/l10n_es_edi_facturae_certificate.py

```diff
@@ -0,0 +1,121 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import datetime
+from base64 import b64decode
+from copy import deepcopy
+import logging
+from unittest.mock import patch
+
+from cryptography.hazmat.primitives.serialization import pkcs12
+from lxml import etree
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools import file_open, ormcache
+
+__logger = logging.getLogger(__name__)
+try:
+    import xmlsig
+    import xades
+except ImportError as err:
+    __logger.debug(err)
+
+class Certificate(models.Model):
+    _name = 'l10n_es_edi_facturae.certificate'
+    _description = 'Facturae Digital Certificate'
+    _order = 'date_start desc, id desc'
+    _rec_name = 'serial_number'
+
+    content = fields.Binary(string="PFX Certificate", required=True, help="PFX Certificate")
+    password = fields.Char(help="Passphrase for the PFX certificate")
+    serial_number = fields.Char(readonly=True, index=True, help="The serial number to add to electronic documents")
+    date_start = fields.Datetime(readonly=True, help="The date on which the certificate starts to be valid")
+    date_end = fields.Datetime(readonly=True, help="The date on which the certificate expires")
+    company_id = fields.Many2one(comodel_name='res.company', default=lambda self: self.env.company, required=True, readonly=True)
+    is_valid = fields.Boolean(string="is the certificate date valid", compute='_compute_is_valid', store=False, readonly=True)
+
+    @api.depends('date_start', 'date_end')
+    def _compute_is_valid(self):
+        for certificate in self:
+            certificate.is_valid = certificate.date_start < fields.Datetime.now() < certificate.date_end
+
+    @ormcache('self.content', 'self.password')
+    def _decode_certificate(self):
+        """
+        Return some certificate data concerning its validity
+        :return tuple: encoded certificate, private key, decrypted certificate
+        """
+        self.ensure_one()
+        _unused, certificate, *_unused = pkcs12.load_key_and_certificates(b64decode(self.content), self.password.encode())
+        return certificate.not_valid_before, certificate.not_valid_after, certificate.serial_number
+
+    # -------------------------------------------------------------------------
+    # LOW-LEVEL METHODS
+    # -------------------------------------------------------------------------
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        certificates = super().create(vals_list)
+        for certificate in certificates:
+            try:
+                cert_date_start, cert_date_end, serial_number = certificate._decode_certificate()
```

Comment:
No need for ormcache if it is only used in the create method.  We are mixing up different things from other modules, but normally we do such method to get the private/public key/... when we need to sign.